app/pages/EditBreakingNewsPage.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useEffect } from 'react';
+import { RiAddBoxLine, RiDeleteBin5Line, RiSave3Line } from 'react-icons/ri';
+
+import * as dataService from '../utils/DataService';
+import { NewsArticle } from 'models';
+import './EditBreakingNewsPage.scss';
+
+export const EditBreakingNewsPage = () => {
+  const [breakingNewsArticles, setBreakingNewsArticles] = useState<Array<NewsArticle>>([]);
+  useEffect(() => {
+    dataService.get('/api/news_articles').then(({ news_articles }) => {
+      console.log('news_articles', news_articles);
+      setBreakingNewsArticles(news_articles.map((article: NewsArticle) => ({
+        ...article,
+        effective_date: formatDate(article.effective_date),
+        expiration_date: formatDate(article.expiration_date)
+      })));
+    })
+  }, []);
+
+  const formatDate = (date: string): string | null => {
+    return date ? new Date(date).toISOString().split('T')[0] : null;
+  }
+
+  const onAddNew = async () => {
+    const response = await dataService.post('/api/news_articles', {
+      effective_date: new Date().toISOString().split('T')[0]
+    });
+    const { news_article } = await response.json();
+    console.log(news_article);
+
+    setBreakingNewsArticles([
+      ...breakingNewsArticles,
+      {
+        ...news_article,
+        effective_date: formatDate(news_article.effective_date),
+        expiration_date: formatDate(news_article.expiration_date)
+      }
+    ]);
+  }
+
+  const onSave = (articleId: number) => {
+    const article = breakingNewsArticles.find(({ id }: any) => id === articleId);
+    console.log(article);
+    dataService.put(`/api/news_articles/${articleId}`, article);
+  };
+
+  const onDelete = async (articleId: number) => {
+    await dataService.APIDelete(`/api/news_articles/${articleId}`);
+    const updatedNewsArticles = breakingNewsArticles.filter(({ id }: any) => id !== articleId);
+    console.log(updatedNewsArticles);
+
+    setBreakingNewsArticles(updatedNewsArticles);
+  };
+
+  const onFieldChange = (articleId: number, { target }: any) => {
+    const { value, dataset } = target;
+    const { field } = dataset;
+
+    setBreakingNewsArticles(breakingNewsArticles.map((article: any) => {
+      if (article.id === articleId) {
+        return {
+          ...article,
+          [field]: value
+        };
+      }
+      return article;
+    }));
+  }
+
+  const renderForm = (article: NewsArticle, index: number): any => (
```

Comment:
Could you extract this into a separate component? While small functions that return JSX elements are fine, this is quite large and I think it would be more readable to pull it out into a separate component.